### ISSUE TITLE
Error during post cluster up configuration: unable to view kubeconfig file

### ISSUE THREAD

**Issue by naithagoni** [minishift/minishift#2786]
I got the error after minishift start "Error during post cluster up configuration: unable to view kubeconfig file".
I have checked my config file which is in "/.kube/config", there is nothing minishift, openshift cluster, and context created in the config file. I have also checked, the file"<profile_name>_kubeconfig" is existed under "/.minishift/machine/".
I have deleted and re-installed minishift but no luck.

### General information

  * Minishift version:  v1.23.0+91235ee
  * OS: Linux / macOS / Windows: ubuntu 18.04 LTS
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox: KVM

### Steps to reproduce

  1. minishift start (got error)
  2. minishift delete
  3. minishift start (got error)

### Expected
minishift, openshift cluster, and context should be added to my config file

### Actual
There is nothing minishift, openshift cluster, and context created in the config file. So, I can not use the minishift as a "use-context".

### Logs
-- minishift version: v1.23.0+91235ee
-- Starting profile 'myproject'
Found binary path at /usr/local/bin/docker-machine-driver-kvm
Launching plugin server for driver kvm
Plugin server listening at address 127.0.0.1:38839
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(myproject) Calling .GetState
(myproject) DBG | Getting current state...
(myproject) DBG | Fetching VM...
The 'myproject' VM is already running.

I have created "myproject" profile but I can not switch to this profile because there is nothing added to my ~/.kube/config file.

**Comment by naithagoni** [comment]
@praveenkumar updated

**Comment by praveenkumar** [comment]
@naithagoni as per logs I can see you have working with `myproject` profile then can you do below and see if that works. From the logs I am not able to figure out then when you deleted does it delete default profile vm or from the `myproject` profile.

```
$ minishift profile delete myproject
$ minishift start --profile myproject
```

**Comment by naithagoni** [comment]
I have solved it by re-installing kubectl and minikube.

**Comment by cmoulliard** [comment]
here is the scenario that I followed using okd 3.10 and minishift v1.23

```
rm -rf ~/.minishift
minishift config set memory 6000
No Minishift instance exists. New 'memory' setting will be applied on next 'minishift start'

minishift addons enable admin-user
Add-on 'admin-user' enabled


MINISHIFT_ENABLE_EXPERIMENTAL=y minishift start --extra-clusterup-flags="--enable=service-catalog,automation-service-broker"
-- Starting profile 'minishift'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.10.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.10.0' is supported ... OK
-- Checking if requested hypervisor 'xhyve' is supported on this platform ... OK
-- Checking if xhyve driver is installed ...
   Driver is available at /usr/local/bin/docker-machine-driver-xhyve
   Checking for setuid bit ... OK
-- Checking the ISO URL ... OK
-- Downloading OpenShift binary 'oc' version 'v3.10.0'
 23.81 MiB / 23.81 MiB [==============================================================================================================================================================================] 100.00% 0s-- Downloading OpenShift v3.10.0 checksums ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'xhyve' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    5 GB
   vCPUs :    2
   Disk size: 20 GB

   Downloading ISO 'https://github.com/minishift/minishift-centos-iso/releases/download/v1.12.0/minishift-centos7.iso'
 344.00 MiB / 344.00 MiB [============================================================================================================================================================================] 100.00% 0s
-- Starting Minishift VM .................. OK
-- Checking for IP address ... OK
-- Checking for nameservers ... OK
-- Checking if external host is reachable from the Minishift VM ...
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ...
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 1% used OK
   Importing 'openshift/origin:v3.10.0' . CACHE MISS
   Importing 'openshift/origin-docker-registry:v3.10.0' . CACHE MISS
   Importing 'openshift/origin-haproxy-router:v3.10.0' . CACHE MISS
-- OpenShift cluster will be configured with ...
   Version: v3.10.0
-- Pulling the Openshift Container Image ................................ OK
-- Copying oc binary from the OpenShift container image to VM ... OK
-- Starting OpenShift cluster .-- Extra 'oc' cluster up flags (experimental) ...
   '--enable=service-catalog,automation-service-broker'
............................................................................................Error during 'cluster up' execution: Error starting the cluster. ssh command error:
command : /var/lib/minishift/bin/oc cluster up --public-hostname 192.168.65.12 --routing-suffix 192.168.65.12.nip.io --base-dir /var/lib/minishift/base --enable=service-catalog,automation-service-broker
err     : exit status 1
output  : Getting a Docker client ...
Checking if image openshift/origin-control-plane:v3.10 is available ...
Pulling image openshift/origin-control-plane:v3.10
Image pull complete
Pulling image openshift/origin-cli:v3.10
Pulled 1/4 layers, 26% complete
Pulled 2/4 layers, 69% complete
Pulled 3/4 layers, 80% complete
Pulled 4/4 layers, 100% complete
Extracting
Image pull complete
Pulling image openshift/origin-node:v3.10
Pulled 5/6 layers, 84% complete
Pulled 5/6 layers, 94% complete
Pulled 6/6 layers, 100% complete
Extracting
Image pull complete
Checking type of volume mount ...
Determining server IP ...
Using public hostname IP 192.168.65.12 as the host IP
Checking if OpenShift is already running ...
Checking for supported Docker version (=>1.22) ...
Checking if insecured registry is configured properly in Docker ...
Checking if required ports are available ...
Checking if OpenShift client is configured properly ...
Checking if image openshift/origin-control-plane:v3.10 is available ...
Starting OpenShift using openshift/origin-control-plane:v3.10 ...
I0912 11:43:50.704935    2100 config.go:42] Running "create-master-config"
I0912 11:43:53.914220    2100 config.go:46] Running "create-node-config"
I0912 11:43:55.260114    2100 flags.go:30] Running "create-kubelet-flags"
I0912 11:43:55.774912    2100 run_kubelet.go:48] Running "start-kubelet"
I0912 11:43:55.942933    2100 run_self_hosted.go:172] Waiting for the kube-apiserver to be ready ...
I0912 11:45:17.954979    2100 interface.go:26] Installing "kube-proxy" ...
I0912 11:45:17.955005    2100 interface.go:26] Installing "kube-dns" ...
I0912 11:45:17.955015    2100 interface.go:26] Installing "openshift-apiserver" ...
I0912 11:45:17.955073    2100 apply_template.go:83] Installing "openshift-apiserver"
I0912 11:45:17.956347    2100 apply_template.go:83] Installing "kube-proxy"
I0912 11:45:17.957767    2100 apply_template.go:83] Installing "kube-dns"
I0912 11:45:21.188019    2100 interface.go:41] Finished installing "kube-proxy" "kube-dns" "openshift-apiserver"
I0912 11:46:18.336011    2100 run_self_hosted.go:224] openshift-apiserver available
I0912 11:46:18.336057    2100 interface.go:26] Installing "openshift-controller-manager" ...
I0912 11:46:18.336116    2100 apply_template.go:83] Installing "openshift-controller-manager"
I0912 11:46:21.148958    2100 interface.go:41] Finished installing "openshift-controller-manager"
Adding default OAuthClient redirect URIs ...
Adding service-catalog ...
Adding automation-service-broker ...
I0912 11:46:21.173044    2100 interface.go:26] Installing "openshift-service-catalog" ...
I0912 11:46:21.173055    2100 interface.go:26] Installing "automation-service-broker" ...
I0912 11:46:21.178806    2100 apply_template.go:83] Installing "automation-service-broker"
I0912 11:46:21.240780    2100 apply_template.go:83] Installing "service-catalog"
E0912 11:51:25.859924    2100 interface.go:34] Failed to install "automation-service-broker": timed out waiting for the condition
I0912 11:51:25.860009    2100 interface.go:41] Finished installing "openshift-service-catalog" "automation-service-broker"
Error: timed out waiting for the condition

minishift stop
Stopping the OpenShift cluster...
Cluster stopped.

minishift start
-- Starting profile 'minishift'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.10.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.10.0' is supported ... OK
-- Checking if requested hypervisor 'xhyve' is supported on this platform ... OK
-- Checking if xhyve driver is installed ...
   Driver is available at /usr/local/bin/docker-machine-driver-xhyve
   Checking for setuid bit ... OK
-- Checking the ISO URL ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'xhyve' hypervisor ...
-- Starting Minishift VM ................. OK
-- Checking for IP address ... OK
-- Checking for nameservers ... OK
-- Checking if external host is reachable from the Minishift VM ...
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ...
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 32% used OK
-- OpenShift cluster will be configured with ...
   Version: v3.10.0
-- Copying oc binary from the OpenShift container image to VM ... OK
-- Starting OpenShift cluster ...................................................
Getting a Docker client ...
Checking if image openshift/origin-control-plane:v3.10 is available ...
Checking type of volume mount ...
Determining server IP ...
Using public hostname IP 192.168.65.12 as the host IP
Checking if OpenShift is already running ...
Checking for supported Docker version (=>1.22) ...
Checking if insecured registry is configured properly in Docker ...
Checking if required ports are available ...
Checking if OpenShift client is configured properly ...
Checking if image openshift/origin-control-plane:v3.10 is available ...
Starting OpenShift using openshift/origin-control-plane:v3.10 ...
I0912 12:19:58.389276    1995 flags.go:30] Running "create-kubelet-flags"
I0912 12:19:59.087069    1995 run_kubelet.go:48] Running "start-kubelet"
I0912 12:19:59.292496    1995 run_self_hosted.go:172] Waiting for the kube-apiserver to be ready ...
I0912 12:20:17.304603    1995 interface.go:26] Installing "kube-proxy" ...
I0912 12:20:17.305176    1995 interface.go:26] Installing "kube-dns" ...
I0912 12:20:17.305182    1995 interface.go:26] Installing "openshift-apiserver" ...
I0912 12:20:17.305223    1995 apply_template.go:83] Installing "openshift-apiserver"
I0912 12:20:17.306865    1995 apply_template.go:83] Installing "kube-proxy"
I0912 12:20:17.307942    1995 apply_template.go:83] Installing "kube-dns"
I0912 12:20:25.826502    1995 interface.go:41] Finished installing "kube-proxy" "kube-dns" "openshift-apiserver"
I0912 12:20:38.866291    1995 run_self_hosted.go:224] openshift-apiserver available
I0912 12:20:38.866322    1995 interface.go:26] Installing "openshift-controller-manager" ...
I0912 12:20:38.866339    1995 apply_template.go:83] Installing "openshift-controller-manager"
I0912 12:20:41.056760    1995 interface.go:41] Finished installing "openshift-controller-manager"
Adding default OAuthClient redirect URIs ...
Adding sample-templates ...
Adding web-console ...
Adding centos-imagestreams ...
Adding router ...
Adding persistent-volumes ...
Adding registry ...
I0912 12:20:41.084090    1995 interface.go:26] Installing "sample-templates" ...
I0912 12:20:41.084104    1995 interface.go:26] Installing "openshift-web-console-operator" ...
I0912 12:20:41.084109    1995 interface.go:26] Installing "centos-imagestreams" ...
I0912 12:20:41.084113    1995 interface.go:26] Installing "openshift-router" ...
I0912 12:20:41.084117    1995 interface.go:26] Installing "persistent-volumes" ...
I0912 12:20:41.084123    1995 interface.go:26] Installing "openshift-image-registry" ...
I0912 12:20:41.084665    1995 interface.go:26] Installing "sample-templates/mongodb" ...
I0912 12:20:41.084675    1995 interface.go:26] Installing "sample-templates/dancer quickstart" ...
I0912 12:20:41.084680    1995 interface.go:26] Installing "sample-templates/django quickstart" ...
I0912 12:20:41.084685    1995 interface.go:26] Installing "sample-templates/nodejs quickstart" ...
I0912 12:20:41.084691    1995 interface.go:26] Installing "sample-templates/jenkins pipeline ephemeral" ...
I0912 12:20:41.084696    1995 interface.go:26] Installing "sample-templates/mariadb" ...
I0912 12:20:41.084700    1995 interface.go:26] Installing "sample-templates/mysql" ...
I0912 12:20:41.084703    1995 interface.go:26] Installing "sample-templates/postgresql" ...
I0912 12:20:41.084714    1995 interface.go:26] Installing "sample-templates/cakephp quickstart" ...
I0912 12:20:41.084728    1995 interface.go:26] Installing "sample-templates/rails quickstart" ...
I0912 12:20:41.084732    1995 interface.go:26] Installing "sample-templates/sample pipeline" ...
I0912 12:20:41.084789    1995 apply_list.go:68] Installing "sample-templates/sample pipeline"
I0912 12:20:41.085137    1995 apply_template.go:83] Installing "openshift-web-console-operator"
I0912 12:20:41.085355    1995 apply_list.go:68] Installing "centos-imagestreams"
I0912 12:20:41.095822    1995 apply_list.go:68] Installing "sample-templates/mysql"
I0912 12:20:41.096070    1995 apply_list.go:68] Installing "sample-templates/mongodb"
I0912 12:20:41.096242    1995 apply_list.go:68] Installing "sample-templates/dancer quickstart"
I0912 12:20:41.096426    1995 apply_list.go:68] Installing "sample-templates/django quickstart"
I0912 12:20:41.096580    1995 apply_list.go:68] Installing "sample-templates/nodejs quickstart"
I0912 12:20:41.096723    1995 apply_list.go:68] Installing "sample-templates/jenkins pipeline ephemeral"
I0912 12:20:41.096874    1995 apply_list.go:68] Installing "sample-templates/mariadb"
I0912 12:20:41.097228    1995 apply_list.go:68] Installing "sample-templates/postgresql"
I0912 12:20:41.097384    1995 apply_list.go:68] Installing "sample-templates/cakephp quickstart"
I0912 12:20:41.097550    1995 apply_list.go:68] Installing "sample-templates/rails quickstart"
I0912 12:20:46.925157    1995 interface.go:41] Finished installing "sample-templates/mongodb" "sample-templates/dancer quickstart" "sample-templates/django quickstart" "sample-templates/nodejs quickstart" "sample-templates/jenkins pipeline ephemeral" "sample-templates/mariadb" "sample-templates/mysql" "sample-templates/postgresql" "sample-templates/cakephp quickstart" "sample-templates/rails quickstart" "sample-templates/sample pipeline"
I0912 12:23:18.460045    1995 interface.go:41] Finished installing "sample-templates" "openshift-web-console-operator" "centos-imagestreams" "openshift-router" "persistent-volumes" "openshift-image-registry"
Login to server ...
Creating initial project "myproject" ...
Server Information ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.65.12:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin


Could not set oc CLI context for 'minishift' profile: Error during setting 'minishift' as active profile: The specified path to the kube config '/Users/dabou/.minishift/machines/minishift_kubeconfig' does not exist
```

**Comment by karlnicholas** [comment]
@praveenkumar  I think you hit the nail on the head with your comment. I too was consistently having this problem but I was able to get it working. This on Windows 10 - CDK 3.7-02. When I installed with minishift the first time I was getting a failure. For me it was failing during the registration process. So I simply stopped minishift and started it again and it went through the second time okay, except for this 'minishift_kubeconfig' issue. So on your comment I deleted everything and this time started it with "minishift start --skip-registration --profile minishift". Skipping registration means it installed completely the first time without errors. I added --profile minishift also because that seemed like a good idea. Now I have a minishift_kubeconfig file and a new .kube directory.
